daml-lf/tests/MultiKeys.daml

```diff
@@ -0,0 +1,232 @@
+-- Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module MultiKeys where
+
+import DA.Assert
+import DA.Exception (throw)
+import DA.Optional
+
+exception E
+  where
+    message "E"
+
+template Keyed
+  with
+    p : Party
+  where
+    signatory p
+    key p : Party
+    maintainer key
+
+-- All these tests operate under the assumption that there
+-- exists two global contracts with the same key `p` and one
+-- of those will always be returned by fetch/lookp-by-key at
+-- the beginning of the choice.
+-- All choices here are accepted in non-uck mode.
+-- Behavior in uck-mode is documented in comments.
```

Comment:
What is "uck-mode" referring to here? Canton's uck mode will reject some of the choices as part of its contract key conflict detection. So is there a "Daml engine uck mode"?

Reply:
clarified the comment.